Sources/SwiftParser/IsValidIdentifier.swift

```diff
@@ -0,0 +1,126 @@
+public enum IdentifierCheckContext {
+  /// Check if a name can be used to declare a variable, ie. if it can be used after a `let` or `var` keyword.
+  ///
+  /// ### Examples
+  /// - `test` is a valid variable name and `let test: Int` is valid Swift code
+  /// - `class` is not a valid variable and `let class: Int` is invalid Swift code
+  case variableName
+
+  /// Check if a name can be used as a member access, ie. if it can be used after a `.`.
+  ///
+  /// ### Examples
+  /// - `test` is a valid identifier for member access because `myStruct.test` is valid
+  /// - `class` is a valid identifier for member access because `myStruct.class` is valid, even though `class`
+  ///   needs to be wrapped in backticks when used to declare a variable.
+  /// - `self` is not a valid identifier for member access because `myStruct.self` does not access a member named
+  ///   `self` on `myStruct` and instead returns `myStruct` itself.
+  case memberAccess
+}
+
+/// Checks whether `name` can be used as an identifier in a certain context.
+///
+/// If the name cannot be used as an identifier in this context, it needs to be escaped.
+///
+/// For example, `class` is not a valid identifier for a variable name and needs to be be wrapped in backticks
+/// to be valid Swift code, like the following.
+///
+/// ```swift
+/// let `class`: String
+/// ```
+///
+/// The context is important here – some names can be used as identifiers in some contexts but not others.
+/// For example, `myStruct.class` is valid without adding backticks `class`, but as mentioned above,
+/// backticks need to be added when `class` is used as a variable name.
+///
+/// - SeeAlso: ``SwiftParser/IdentifierCheckContext``
+public func isValidIdentifier(_ name: String, for context: IdentifierCheckContext) -> Bool {
+  switch context {
+  case .variableName:
+    return isValidVariableName(name)
+  case .memberAccess:
+    return isValidMemberAccess(name)
+  }
+}
+
+private func isValidVariableName(_ name: String) -> Bool {
+  var parser = Parser("var \(name)")
+  let decl = DeclSyntax.parse(from: &parser)
+  guard parser.at(.endOfFile) else {
+    // We didn't parse the entire name. Probably some garbage left in the name, so not an identifier.
+    return false
+  }
+  guard !decl.hasError && !decl.hasWarning else {
+    // There were syntax errors in the source code. So not valid.
+    return false
+  }
+  guard let variable = decl.as(VariableDeclSyntax.self) else {
+    return false
+  }
+  guard Array(variable.tokens(viewMode: .sourceAccurate)).count == 2 else {
+    // We expect to parse exactly two tokens ('var' and the name).
+    // If we parsed more, the name got parsed as two tokens and is thus not a
+    // valid name.
+    return false
+  }
+  guard let identifier = variable.bindings.first?.pattern.as(IdentifierPatternSyntax.self)?.identifier else {
+    return false
+  }
+  guard identifier.tokenKind == .identifier(name) else {
```

Comment:
```suggestion
  guard identifier.rawTokenKind == .identfiier && identifier.rawTokenText.count == name.utf8.count else {
```
should give us a _tiny_ performance increase.